src/indexes/noop.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Captor,
+  Extension,
+  IndexableObj,
+} from './common';
+
+
+export function createNoopCaptor<T extends IndexableObj>(): Captor<T> {
+  const captureObject: Captor<T> = (obj: T): T & Extension<T> => {
+    const extendedObject = Object.assign(obj, {
+      deleteFromIndex() {
+        // No-op
+      },
+      getTarget() {
+        const {getTarget, deleteFromIndex, ...rest} = extendedObject;
+        return rest;
+      }
+    });
+
+    return extendedObject;
+  };
+
+  return captureObject;
+}
```

Comment:
```suggestion
export function createNoopCaptor<T extends IndexableObj>(): Captor<T> {
  const captureObject: Captor<T> = (obj: T): T & Extension<T> => {
    const extendedObject = Object.assign(obj, {
      deleteFromIndex() {
        // No-op
      },
      getTarget() {
        return obj;
      }
    });

    return extendedObject;
  };

  return captureObject;
}
```

would we want to return the original object? then we don't create a new object reference and incur very little overhead since we don't need to copy any properties. this will make it much more efficient when being sent by `postMessage` across IPC

Reply:
I think we can now return the original object, now that getTarget() can safely return T.

Reply:
did we want to add an `isProxied` method to the extended object?

Reply:
Oh actually, I remember I tried that. The issue is that `Object.assign` modifies the target object. That means if we return obj, it will still have `deleteFromIndex` and `getTarget`.

Reply:
I could do something like this.
```
    const extendedObject = Object.assign({}, obj, {
      deleteFromIndex() {
        deleters.forEach((deleteFromIndexFunc) => deleteFromIndexFunc(obj));
      },
      getTarget() {
        return obj;
      }
    });
```

Reply:
instantiating another object and copying properties over is a surprisingly expensive operation -- it's definitely worth a few more lines of code to prevent instantiating another object and populating it